wsgi.py

```diff
@@ -32,24 +28,57 @@ def wake_up():
     ai_service_id = input_data.get('ai_service', 'generic_ai')
     raw_data = input_data['data']
 
-    s3_destination = f'{AWS_BUCKET}/{data_id}/{ai_service_id}'
-
-    application.logger.info(
-        'Saving data to location:  s3://%s', s3_destination
-    )
-
-    filesystem = s3.connect(AWS_KEY, AWS_SECRET)
-    s3.save_data(filesystem, s3_destination, raw_data)
-
-    message = {
-        'message': f'AI-Ops pipeline successfull for {data_id}',
-        'url': s3_destination
+    try:
+        with tarfile.open(fileobj=tempfile.NamedTemporaryFile(delete=False), mode='w:gz') as f:   # noqa
+            data = io.BytesIO(json.dumps(raw_data).encode())
+            info = tarfile.TarInfo(name=f'{ai_service_id}_{data_id}.json')
+            info.size = len(data.getvalue())
+            temp_file_name = f.name
+            f.addfile(info, data)
+
+    except Exception as e:    # noqa
+        error_msg = 'Error during TAR.GZ creation: ' + str(e)
+        ROOT_LOGGER.exception("Exception: %s", error_msg)
+        return jsonify(
+            status='Error',
+            type=str(e.__class__.__name__),
+            message=error_msg
+        ), 500
+
+    files = {
+        'upload': (
+            temp_file_name, open(temp_file_name, 'rb'),
+            'application/vnd.redhat.aiopspublisher.aiservice+tgz'
```

Comment:
If the AI Service ID had dashes/underscores, upload won't work. There is a regex in the upload code that will block the upload because of that
`content_regex = r'^application/vnd\.redhat\.([a-z]+)\.([a-z]+)\+(tgz|zip)$'`

Hence, it needs to be hardcoded

Reply:
So what about stripping the illegal chars away? I mean.. This string is intended for the consumer of the data, correct? So the consumer can know who's the sender. Once we have multiple AI services on board, how would our consumer know, which message is relevant to him, since it's interested in one particular type of AI processing and not the others?

Reply:
Ok, so some good news on this front...

The Upload folks will be working on a PR to make the regex less restrictive.
The new regex will allow lowercase letters, numbers and hyphens -- that should be enough for us to include the AI service ID here

Reply:
Let's hold off on making changes here until the regex change has been made official in the Upload repo. We should know early next week.